mobile/scripts/buildApp.sh

```diff
@@ -109,17 +110,136 @@ if [[ "${OS}" == "android" ]]; then
   # Compile the app
   xcodebuild -configuration Release -target Status install -sdk "$SDK" -arch "$ARCH" DSTROOT="$BIN_DIR" INSTALL_PATH="/" TARGET_BUILD_DIR="$BIN_DIR" CODE_SIGN_IDENTITY="" CODE_SIGNING_REQUIRED=NO CODE_SIGNING_ALLOWED=NO | xcbeautify
 
-  if [[ -e "$BIN_DIR/Status.app/Info.plist" ]]; then
-    echo "Build succeeded"
-  else
+  if [[ ! -e "$BIN_DIR/Status.app/Info.plist" ]]; then
     echo "Build failed"
     exit 1
   fi
+
+  if [[ "$SIGN_IOS" == "true" ]]; then
+    echo "Signing iOS app..."
+
+    if [[ -z "$IOS_CERT_PATH" || -z "$IOS_CERT_PASSWORD" || -z "$IOS_PROVISIONING_PROFILE" ]]; then
+      echo "Error: Missing iOS signing credentials"
+      exit 1
+    fi
+
+    KEYCHAIN_NAME="build.keychain"
+    KEYCHAIN_PASSWORD=$(openssl rand -base64 16)
+
+    cleanup_keychain() {
+      echo "Cleaning up keychain..."
+      security default-keychain -s login.keychain 2>/dev/null || true
+      security delete-keychain "$KEYCHAIN_NAME" 2>/dev/null || true
+    }
+
+    trap cleanup_keychain EXIT
+
+    security delete-keychain "$KEYCHAIN_NAME" 2>/dev/null || true
+
+    security create-keychain -p "$KEYCHAIN_PASSWORD" "$KEYCHAIN_NAME"
+    security unlock-keychain -p "$KEYCHAIN_PASSWORD" "$KEYCHAIN_NAME"
+    security set-keychain-settings -t 3600 -u "$KEYCHAIN_NAME"
+    security list-keychains -s "$KEYCHAIN_NAME" login.keychain
+    security default-keychain -s "$KEYCHAIN_NAME"
+
+    echo "Importing Apple WWDR G3 certificate..."
+    WWDR_TEMP_DIR=$(mktemp -d)
+    curl -sS -o "$WWDR_TEMP_DIR/AppleWWDRCAG3.cer" https://www.apple.com/certificateauthority/AppleWWDRCAG3.cer
+    security import "$WWDR_TEMP_DIR/AppleWWDRCAG3.cer" -k "$KEYCHAIN_NAME" -T /usr/bin/codesign
+    rm -rf "$WWDR_TEMP_DIR"
+    echo "Apple WWDR G3 certificate imported"
+
+    security import "$IOS_CERT_PATH" -k "$KEYCHAIN_NAME" -P "$IOS_CERT_PASSWORD" -T /usr/bin/codesign
+    security set-key-partition-list -S apple-tool:,apple: -s -k "$KEYCHAIN_PASSWORD" "$KEYCHAIN_NAME"
+
+    PROFILE_DIR="$HOME/Library/MobileDevice/Provisioning Profiles"
+    mkdir -p "$PROFILE_DIR"
+
+    PROFILE_UUID=$(security cms -D -i "$IOS_PROVISIONING_PROFILE" 2>/dev/null | grep -A1 "<key>UUID</key>" | grep "<string>" | sed 's/.*<string>\(.*\)<\/string>.*/\1/')
+
+    rm -f "$PROFILE_DIR/$PROFILE_UUID.mobileprovision"
+
+    cp "$IOS_PROVISIONING_PROFILE" "$PROFILE_DIR/$PROFILE_UUID.mobileprovision"
+
+    echo "Installed provisioning profile: $PROFILE_UUID"
+
+    echo "Embedding provisioning profile into app..."
+    cp "$IOS_PROVISIONING_PROFILE" "$BIN_DIR/Status.app/embedded.mobileprovision"
+
+    echo "Searching for signing identity in keychain..."
+    security find-identity -v -p codesigning "$KEYCHAIN_NAME"
+
+    SIGNING_IDENTITY=$(security find-identity -v -p codesigning "$KEYCHAIN_NAME" | grep -E "iPhone Distribution|Apple Distribution" | head -1 | awk '{print $2}')
```

Comment:
Are there really no good tools to do signing of OSX apps? Why do we write our own tooling for this?

Reply:
There is `fastlane`, but I've had to perform similar hacks there too in status-mobile so I figured we're better off with our own bash script.

ref : https://github.com/status-im/status-mobile/commit/cd6ec302ad8de6f906541b72f2517e1ead76b91f